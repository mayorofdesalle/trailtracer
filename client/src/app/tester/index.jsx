import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
