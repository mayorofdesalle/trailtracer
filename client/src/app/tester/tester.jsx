import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };