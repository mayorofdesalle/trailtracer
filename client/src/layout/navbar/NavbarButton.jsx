import PropTypes from 'prop-types';
import { useRef } from 'react';
import styled from 'styled-components';

import { Button, Icon } from '@components/ui';
import { useHover } from '@hooks';

const NavbarButtonInner = styled(Button)`
    flex-shrink: 0;
    padding: 0;
    width: clamp(3rem, min(6dvw, 6dvh), 12rem);
    height: clamp(1.5rem, min(3dvw, 3dvh), 3rem);

    &:active {
        background-color: ${({ theme }) => theme.colors.secondary};
    }

    & > svg {
        width: 100%;
        height: 100%;
    }
`;

const NavbarButton = ({ icon, onClick, ...props }) => {
    const button = useRef();
    const isHovered = useHover(button);

    return (
        <NavbarButtonInner ref={button} onClick={onClick} {...props}>
            <Icon name={isHovered ? `${icon}-fill` : `${icon}-line`} />
        </NavbarButtonInner>
    );
};

NavbarButton.propTypes = {
    icon: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    props: PropTypes.object
};

export default NavbarButton;