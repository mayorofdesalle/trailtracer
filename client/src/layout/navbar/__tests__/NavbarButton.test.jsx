import { describe, expect, it, render, screen, vi } from '@app/tester';

import NavbarButton from '../NavbarButton';

describe('NavbarButton', () => {

    it('should render correctly', () => {
        render(<NavbarButton onClick={vi.fn()} icon='forbid' />);
        const button = screen.getByRole('button');
        expect(button).toBeInTheDocument();
    });

});