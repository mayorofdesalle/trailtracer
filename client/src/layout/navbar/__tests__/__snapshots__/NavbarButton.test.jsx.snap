// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavbarButton > should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-beySPh sc-kFCroH zbkUW kgqzMY"
    type="button"
  >
    <svg
      class="sc-eDLKkx fuXsHX"
      fill="currentColor"
      role="img"
    >
      <title>
        forbid-line
      </title>
      <use
        href="http://localhost:3000/src/assets/icons.svg#forbid-line"
      />
    </svg>
  </button>
</DocumentFragment>
`;
